src/addrman.h

```diff
@@ -679,9 +692,9 @@ friend class CAddrManTest;
         CAddrInfo ret;
```

Comment:
It feels like this is a good opportunity to make the access patterns consistent here:

```diff
@@ -689,13 +689,10 @@ public:
     //! Randomly select an address in tried that another address is attempting to evict.
     CAddrInfo SelectTriedCollision()
     {
-        CAddrInfo ret;
-        {
-            LOCK(cs);
-            CheckWithLockHeld();
-            ret = SelectTriedCollision_();
-            CheckWithLockHeld();
-        }
+        LOCK(cs);
+        CheckWithLockHeld();
+        CAddrInfo ret = SelectTriedCollision_();
+        CheckWithLockHeld();
         return ret;
     }
 
@@ -704,26 +701,21 @@ public:
      */
     CAddrInfo Select(bool newOnly = false)
     {
-        CAddrInfo addrRet;
-        {
-            LOCK(cs);
-            CheckWithLockHeld();
-            addrRet = Select_(newOnly);
-            CheckWithLockHeld();
-        }
+        LOCK(cs);
+        CheckWithLockHeld();
+        CAddrInfo addrRet = Select_(newOnly);
+        CheckWithLockHeld();
         return addrRet;
     }
 
     //! Return a bunch of addresses, selected at random.
     std::vector<CAddress> GetAddr(size_t max_addresses, size_t max_pct)
     {
-        Check();
+        LOCK(cs);
+        CheckWithLockHeld();
         std::vector<CAddress> vAddr;
-        {
-            LOCK(cs);
-            GetAddr_(vAddr, max_addresses, max_pct);
-        }
-        Check();
+        GetAddr_(vAddr, max_addresses, max_pct);
+        CheckWithLockHeld();
         return vAddr;
     }
```

(i.e. for all of these functions take the lock, run Check(), run the inner function, run Check() again, release lock/return)

Reply:
:+1: This would make the `Check()` function unused, so it can be removed.

Reply:
Thanks! Done in e21c0dd57009197dfeedbc4ed4d30454b7a05cf6.